### ISSUE TITLE
Implement _include_entity for all source collectors

### ISSUE THREAD

**Issue by wkoot** [ICTU/quality-time#5240]
# User story

Handle the open points from #4714, as listed in #5221

# Tasks

## Collector

- [x] AxeAccessibilityCollector checks violations on impact and tags
- [x] AzureDevopsJobs + AzureDevopsPipelines do include checks on Job, before making Entities
- [x] AzureDevopsIssues do include checks on work item, before making Entities
- [x] what about TransactionEntity.is_to_be_included and SlowTransactionsCollector._is_to_be_included_and_is_slow ?
- [x] GitLabJobsBase does include checks on Job, before making Entities
- [x] GitLabPipelineUpToDateness checks pipeline status and trigger, but only returns aggregate
- [x] JenkinsJobRunsWithinTimePeriod has both _include_job (from parent) and its own _include_build
- [x] JenkinsJobs yields jobs based on include checks on Job, before making Entities
- [x] JiraIssues _include_issue omits items which cannot be parsed by current logic, which needs to be implemented
- [x] QualityTimeMetrics sneakily also does __update_entity when an entity is selected for inclusion
- [x] ~~TestNGTests checks both beforeClass/afterClass and has complicated test_results_to_count logic~~
- [x] TrelloBase implementations have a different structure but both use a __ignore_card
